### ISSUE TITLE
Pydantic BaseModel raises too-few-public-methods warning

### ISSUE THREAD

**Issue by robinderat** [PyCQA/pylint#7235]
### Bug description

I am using a class extending BaseModel from pydantic to store some data. 

```
from pydantic import BaseModel
class Keyword(BaseModel):
    foo: str
    bar: int
```
Pylint gives a too-few-public-methods warning since the class does not have/need any public methods. So far so good. The exclude-too-few-public-methods parameter was added to alleviate exactly this problem https://github.com/PyCQA/pylint/issues/3370, but it doesn't work.

I have tried putting it in the pyproject.toml 
```
[tool.pylint]
exclude-too-few-public-methods="pydantic.*"
```

I have tried putting it in a .pylintrc
```
[DESIGN]
exclude-too-few-public-methods=pydantic.*
```

I have tried using it with CLI
```
pylint keyword.py --exclude-too-few-public-methods "pydantic.*"
```

I tried `pydantic.*` , `"pydantic.*"` , `[pydantic.*]`, `["pydantic.*"]`, `"[pydantic.*]"`, `pydantic.main.BaseModel`, `BaseModel`, `Keyword`. It just doesn't work

Either I am severely misunderstanding the documentation, which unfortunately only gives 
```
[tool.pylint.design]
exclude-too-few-public-methods = []
``` 
as an example, or the parameter is just ignored by pylint.

### Configuration

```ini
[MASTER]
extension-pkg-whitelist=pydantic
```

### Command used

```shell
pylint keyword.py
```

### Pylint output

```shell
************* Module bank_transactions_classifier.app.keyword
keyword.py:18:4: R0903: Too few public methods (0/2) (too-few-public-methods)
```

### Expected behavior

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 7.33/10, +2.67)

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.8.10 (v3.8.10:3d8993a744, May  3 2021, 08:55:58)
[Clang 6.0 (clang-600.0.57)]

Note i also tried pylint==2.12.0 where the feature was added
```

### OS / Environment

MacOs 10.15.7 

### Additional dependencies

pydantic==1.9.1

**Comment by mbyrnepr2** [comment]
I've played around with this & I can't reproduce if I use the example provided & a pylintrc containing:
```python
[MASTER]
extension-pkg-whitelist=pydantic
```

When [this line](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/design_analysis.py#L487) is reached, `all_methods` is `13` & hence the message is not emitted.

`ancestor.qname()` is `pydantic.main.BaseModel` on [this line.](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/design_analysis.py#L473) so your regexes `pydantic.*` & `pydantic.main.BaseModel` look appropriate.

FYI I'm on:
```python
pylint 2.15.0-dev0
astroid 2.12.2
Python 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53) 
[Clang 6.0 (clang-600.0.57)]
```

**Comment by mbyrnepr2** [comment]
Without the `extension-pkg-whitelist=pydantic` option, Pylint cannot parse Pydantic's `BaseModel`. This part is similar to the discussion in https://github.com/PyCQA/pylint/issues/7486 which we closed yesterday. 
With this option included, Pylint sees 13 public methods & will not emit `too-few-public-methods` (when using the default value of `min-public-methods=2`):

```python
pylint example.py --min-public-methods=100 --extension-pkg-whitelist=pydantic 
example.py:2:0: R0903: Too few public methods (13/100) (too-few-public-methods)
```

To demonstrate expected behaviour of the `--exclude-too-few-public-methods` option, this one does not emit `too-few-public-methods`:
```python
pylint example.py --min-public-methods=100 --extension-pkg-whitelist=pydantic --exclude-too-few-public-methods=pydantic.*
```

Perhaps that is enough to close the issue but lets say, for example, in the original example & using pylintrc defaults, if `BaseModel` happened to contain only 1 public method, then you would expect Pylint to emit a false positive for `too-few-public-methods` for the `Keyword` class. 
I know that `pylint-pydantic` has an open issue for a similar example, so maybe that is the way forward if a fix is implemented there to supress this warning for these classes. 

@Pierre-Sassoulas since you noticed yesterday's Pydantic issue what do you think - close as won't fix?

**Comment by Pierre-Sassoulas** [comment]
Yes, let's rely on pylint-pydantic for that.
